### PR TITLE
Deprecate rank.tophat and rank.bottomhat

### PR DESCRIPTION
**Pull request by sciunto**
## Description

`rank.tophat()` and `rank.bottomhat()` are two filters for which the behavior does not correspond to the standard nomenclature. We found a little interest in these filters. These filters can be found in `morphology`.

See https://github.com/scikit-image/scikit-image/pull/3614 for the explanations.

## Checklist

- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

File: skimage/filters/rank/__init__.py [hunk 1]
```diff
@@ -12,7 +12,6 @@
 
 __all__ = ['autolevel',
            'autolevel_percentile',
-           'bottomhat',
```
Comment (hmaarrfk) [inline skimage/filters/rank/__init__.py]: should these be removed now?

Comment (sciunto) [reply skimage/filters/rank/__init__.py]: I removed them to facilitate the tests (it uses this variable)I can handle it differently if necessary

Comment (hmaarrfk) [reply skimage/filters/rank/__init__.py]: __all__ helps with autocomplete and introspection. So I would prefer to keep it until we really don't want users to use it.

**Comment by hmaarrfk** [comment]
There seem to be a few warnings that are issued without the user directly calling tophat

```
filters/rank/generic.py::skimage.filters.rank.generic.bottomhat

  /home/travis/venv/lib/python3.7/site-packages/skimage/filters/rank/generic.py:1: FutureWarning: rank.bottomhat is deprecated. This filter is named incorrectly, which can be confusing. As we believe that the usefulness is narrow, this function will be removed in 0.19.

    r"""

filters/rank/generic.py::skimage.filters.rank.generic.tophat

  /home/travis/venv/lib/python3.7/site-packages/skimage/filters/rank/generic.py:1: FutureWarning: rank.tophat is deprecated. This filter is named incorrectly, which can be confusing. As we believe that the usefulness is narrow, this function will be removed in 0.19.

    r"""


```

File: skimage/filters/rank/__init__.py (hunk 1 above)
Comment (emmanuelle) [reply skimage/filters/rank/__init__.py]: Hum, I agree here, if it's in the API (albeit deprecated, it should stay in `__all__`).

**Comment by emmanuelle** [comment]
I will approve once the functions make it back to `__all__` :-)

**Comment by sciunto** [comment]
@hmaarrfk Good catch, but I did not find the reason for this... :/ Do you have any idea?

**Comment by sciunto** [comment]
The ``__all__`` thing is fixed.

**Comment by emmanuelle** [comment]
@hmaarrfk where did you see the spurious warnings? I thought our CI failed on uncaught warnings?

**Comment by hmaarrfk** [comment]
Emmanuelle, our CI does not fail on uncaught warnings. That was decided to be too harsh on PRs. But generally, I look through Travis logs.

**Comment by emmanuelle** [comment]
oh ok, thanks @hmaarrfk

**Comment by emmanuelle** [comment]
The warning is weird... I believe it comes from this import https://github.com/scikit-image/scikit-image/blob/master/skimage/filters/tests/test_median.py#L7 (trying to guess from the position at which the warning occurs, but not 100% sure), but I'm still puzzled that it would trigger the warning since the tophat etc. functions are not called.

**Comment by emmanuelle** [comment]
Anyway this is annoying but it should not block the PR I hope, it's important to get this one on the release wagon since it involves a deprecation.

**Comment by sciunto** [comment]
@emmanuelle It's up to you :)

**Comment by emmanuelle** [comment]
Thanks @sciunto !

**Pull request merged**